### DISCUSSION TITLE
PTR records not created with "ipam-coupling" enabled.

### DISCUSSION THREAD

**Discussion by bitcollector1** [sys4/netbox-plugin-dns#140] [Q&A] [answered]
I'm having an issue where I'm not seeing the PTR records created when using the "ipam-coupling"; I do have the reverse zones defined and they do show up fine when I create the A record manually.  

Not sure exactly what's up and was curious if others have seem something similar? 

Thanks in advance.

**Comment by bitcollector1** [comment]
The fact that the plugin automatically creates the PTR record is very helpful, as I use that data to feed that into my BIND9 configuration file. 

I'm new to DNS and crafting these reverse zone entries by hand was very error prone!

**Comment by peteeckel** [comment]
That's a bit difficult to analyse as, to use the dreaded phrase, "it works for me" ...

Did you by any chance check the "Disable PTR" boolean custom field for the IP address?

**Comment by peteeckel** [comment]
Just to make sure I tested this manually (finding #141 in the process, thanks for making me stumble into this :-))

I can confirm that with NetBox 3.7.0 and NetBox DNS 0.21.15 PTR records are created automatically unless explicitly disabled, provided the PTR zone is available.

**Reply by bitcollector1** [reply]
hmmmm....not sure exactly what's up, as it's still not working for me as far as I can tell. 

I am on NetBox 3.7.1 and Python 3.10........looks like I have some more reading and homework to do ;) 

Thanks again for the plugin/support. 

Cheers!

**Reply by bitcollector1** [reply]
just to be crystal clear,  this is only with the "ipam-coupling" as the plugin works fine to create the PTR's normally.

**Reply by peteeckel** [reply]
No, that's not a bug.

An IP address that is not active is ... well, inactive. As such, no active DNS records are created for it, and so there is no PTR record.

Works as designed, I'd say.

**Comment by bitcollector1** [comment]
I also had a question about the status, it looks like the managed A record that is created via ipam-coupling, is not active. 

<img width="1271" alt="Screenshot 2024-01-19 at 16 25 00" src="https://github.com/peteeckel/netbox-plugin-dns/assets/50723251/522df782-4f8a-492c-b119-3fe931e88be3">

I tried making it active from the API but I was not allowed to do that. 

Finally, you can see a PTR record that was created for me when I did the manual creation of the A record using the plugin GUI.

**Comment by bitcollector1** [comment, marked as answer]
Found the BUG :) 

I have a status of "reserved" for the IP Address.  As soon as I set it to "active" the ptr record was created!  

<img width="863" alt="Screenshot 2024-01-19 at 20 48 04" src="https://github.com/peteeckel/netbox-plugin-dns/assets/50723251/292c1006-2e61-4d20-b60e-bc8c78c08c27">

**Reply by bitcollector1** [reply]
maybe it's not a bug......but it had me stumped!  

one of my team members decided that "reserved" was for IP's in use and "active" means it's available.

**Comment by bitcollector1** [comment]
I'm leaning more towards this is a bug after thinking on this more. The reason I state this is because the PTR record was created for my A record when I did it via GUI......even though the address was "reserved" and NOT in the active state. 

Anyway, I have a special knack for finding things like this, haha......If anyone ever wants to test their plugins they just need to send it my way because I'm sure to turn a knob nobody else thought to ;)

**Reply by peteeckel** [reply]
When you create a DNS record using the GUI, that record is in no way linked to an IP address, even if that IP address happens to match the address record's value. Its default status is `active`, so the PTR gets created.

When you create an IP address and have the address record created automatically, that address record (which is a managed record, i.e. not modifiable via the GUI or API) will match the state of the IP address, which means it will be inactive unless the IP addresses state is active. The rationale behind that is that we don't want to have active DNS records for IP addresses that are only planned, reserved, retired or whatever, just for active IP addresses.

That your team uses the status `active` for inactive IP addresses and `reserved` for active IP addresses is a case that ... well, let's say it isn't very common, and actually quite counterintuitive.

**Reply by bitcollector1** [reply]
agree 100% about the active/reserved status....but it's a battle I just did not want to fight at the time. Funny story......last night I asked my wife her opinion and she came up with the same answer as my team member and then she got offended when I called her thinking back asswards ;)

**Comment by peteeckel** [comment]
... on the other hand ...

While I still find the idea of not using the status 'Active' for active IP addresses somewhat disturbing, there is something important to take home from this discussion. The mapping 'IP Address active -> DNS Record active, else inactive' is actually too unflexible.

First, there are the 'DHCP' and 'SLAAC' statuses that need to be handled as active as well. Whether or not they should create active DNS records may vary, though. Second, NetBox makes it possible to extend the selection list for the IP address status field, and any new status defined that way may or may not require an active DNS record. Currently any new status defined there will result in inactive DNS records.

So I just added FR #144 and PR #143, and released version [0.21.16](https://github.com/peteeckel/netbox-plugin-dns/releases/tag/0.21.16) that contains the new feature.

In your case, the following configuration should be added to `configuration.py`:
```
PLUGINS_CONFIG = {
    'netbox_dns': {
        ...
        'ipam_coupling_ip_address_status_list': [
            'reserved',
            'dhcp',
            'slaac',
        ],
        ...
    },
}
```
This will result in the `active` and `reserved` statuses swapping places :-)

**Comment by bitcollector1** [comment]
Just wanted to say thank you for the latest update!  

We are using DHCP heavily in our lab and will be creating DNS entries for all these systems using your plugin :)
